### PR TITLE
[lsp] added `workspace/executeCommand` support (fixes #425)

### LINKED ISSUE #425: [LSP] Add support for `workspace/executeCommand`.
Languages shall be able to register server commands.

### PR DESCRIPTION
**Pull request by svenefftinge**
Signed-off-by: Sven Efftinge <sven.efftinge@typefox.io>

### CODE REVIEW

File: org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/commands/CommandRegistry.xtend [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+ */
+@Log class CommandRegistry {
+	
+	private static val METHOD = 'workspace/executeCommand'
+	
+	Multimap<String, ICommandsService> registeredCommands
+	
+	LanguageClient client
+	
+	def void initialize(Iterable<? extends IResourceServiceProvider> allLanguages, ClientCapabilities capabilites, LanguageClient client) {
```
Comment (spoenemann) [inline org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/commands/CommandRegistry.xtend]: `capabilites` => `capabilities`

File: org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/commands/CommandRegistry.xtend [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+		]
+	}
+	
+	def Object executeCommand(ExecuteCommandParams params, ILanguageServerAccess access, CancelIndicator cancelIndicator) {
+		var Object result = null
+		for (service : this.registeredCommands.get(params.command)) {
+			val localResult = service.execute(params, access, cancelIndicator)
+			if (localResult !== null) {
+				if (result !== null) {
+					LOG.error('''Multiple commands '�params.command�' have been registered. All are executed but only one result will be send back.''')
```
Comment (spoenemann) [inline org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/commands/CommandRegistry.xtend]: `be send` => `be sent`

File: org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/commands/CommandRegistry.xtend [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+				} else {
+					result = localResult
+				}
+			}
+		}
+		return result
+	}
+	
+	def getCommands() {
+		return this.registeredCommands.keys.toList
```
Comment (spoenemann) [inline org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/commands/CommandRegistry.xtend]: Use `keySet` instead of `keys`.

**Comment by cdietrich** [comment]
lot of red tests

**Pull request merged**